lxd/network_allocations.go

```diff
@@ -0,0 +1,234 @@
+//
+//  Returns a list of network allocations in use by a LXD deployment.
+//
+//  ---
+//  produces:
+//    - application/json
+//  parameters:
+//    - in: query
+//      name: project
+//      description: Project name
+//      type: string
+//      example: default
+//    - in: query
+//      name: all-projects
+//      description: Retrieve entities from all projects
+//      type: boolean
+//  responses:
+//    "200":
+//      description: API endpoints
+//      schema:
+//        type: object
+//        properties:
+//          type:
+//            type: string
+//            description: Response type
+//            example: sync
+//          status:
+//            type: string
+//            description: Status description
+//            example: Success
+//          status_code:
+//            type: integer
+//            description: Status code
+//            example: 200
+//          metadata:
+//            type: array
+//            description: List of network allocations used by a consuming entity
+//            items:
+//              $ref: "#/definitions/NetworkAllocations"
+//    "403":
+//      $ref: "#/responses/Forbidden"
+//    "500":
+//      $ref: "#/responses/InternalServerError"
+func networkAllocationsGet(d *Daemon, r *http.Request) response.Response {
+	var err error
+	projectName := queryParam(r, "project")
+	allProjects := shared.IsTrue(queryParam(r, "all"))
+
+	var projectNames []string
+	err = d.db.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		// Figure out the projects to retrieve.
+		if !allProjects {
+			projectNames = []string{projectName}
+		} else {
+			// Get all project names if no specific project requested.
+			projects, err := dbCluster.GetProjects(ctx, tx.Tx())
+			if err != nil {
+				return fmt.Errorf("Failed loading projects: %w", err)
+			}
+
+			projectNames = make([]string, 0, len(projects))
+			for _, project := range projects {
+				projectNames = append(projectNames, project.Name)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	result := make([]api.NetworkAllocations, 0)
+
+	// Then, get all the networks, their network forwards and their network load balancers.
+	for _, projectName := range projectNames {
+		networkNames, err := d.db.Cluster.GetNetworks(projectName)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		// Get all the networks, their network forwards and their network load balancers.
+		for _, networkName := range networkNames {
+			networkID, _, _, err := d.db.Cluster.GetNetworkInAnyState(projectName, networkName)
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			n, err := network.LoadByName(d.State(), projectName, networkName)
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			// Get the network state.
+			state, err := resources.GetNetworkState(networkName)
```

Comment:
I dont think you need to use this.
Instead use `n.Leases()` the result will include the `gateway` IPs on the network bridge itself.